ngx-style.pl

```diff
@@ -266,6 +269,34 @@
             $need_variable_align = 1;
             $variable_align_space = 0;
         }
+
+        if ($expect_empty_lines) {
+            if ($line =~ m{^\r?\n$}x) {
+                $expect_empty_lines--;
+                $should_not_empty = $expect_empty_lines == 0 ? 1 : 0;
+
+            } else {
+                # ignore vi setting in the end of the file
+                # ignore } followed by a blank line and #endif
+                if ($line !~ m{(\#endif|vi:set)}x) {
+                    output "need to keep two blank lines between functions";
+                }
+
+                $expect_empty_lines = 0;
+                $should_not_empty = 0;
+            }
+
+        } elsif ($should_not_empty) {
+            if ($line =~ m{^\r?\n$}x) {
+                output "too many blank lines between functions";
+            }
+
+            $should_not_empty = 0;
+        }
+
+        if ($line =~ m{^(\}\r?\n)$}x) {
+            $expect_empty_lines = 2;
```

Comment:
This is not necessarily function body ends. It could also be a struct/array initializer...